modules/lang-painless/src/test/resources/rest-api-spec/test/painless/15_update.yml

```diff
@@ -132,7 +132,7 @@
           body:
             script:
               lang:   painless
-              source: "for (def key : params.keySet()) { ctx._source[key] = params[key]}"
+              source: "ctx._source.ctx = ctx"
```

Comment:
Isn't this creating a cycle in _source?

Reply:
@rjernst that's the point, see what it matches a few lines down  and my PR description https://github.com/elastic/elasticsearch/pull/32096/files/049606c19854ac69ee07f2ff5d772f02532f769a#diff-585f6613de46bec6547c7b2aed89db8bR140 :)

Reply:
Ah, thanks, sorry I missed that.